hooks/helpers/useHandleDecimalInput.ts

```diff
@@ -10,13 +11,17 @@ export function useOnChange(
   const { addErrorMessage, removeErrorMessage } = useErrorContext();
 
   return (event: ChangeEvent<HTMLInputElement>) => {
-    const value = event.target.value;
+    let value = event.target.value;
 
     if (type !== "number") {
       onInput(value);
       return;
     }
 
+    if (value.includes(maximumApprovalAmountString)) {
```

Comment:
If the input has the maximum approval amount string in it because that was what was input previously, it will look like "Unlimited approval2" etc. We remove the unlimited approval string and use just the number. This effectively clears the unlimited approval input if the user decides to type a different number instead.